prime_factor/prime_factor.c

```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+void generate_primes(int *ar, int *cnt) {
```

Comment:
Nice way of implementation. 👍 
Though I would suggest that this method returns an `int` instead i.e. the no of primes generated.